### DISCUSSION TITLE
Fixed-length bitvectors

### DISCUSSION THREAD

**Discussion by jp43** [chemosim-lab/ProLIF#160] [Q&A]
Hello, I would like to use ProLIF to generate IFPs to be used as input for ML model predicting pkd. To do that, I would need IFPs as bitvectors of fixed length for different ligand-protein systems. Is there a way to do this with ProLIF?

**Comment by cbouy** [comment]
Hi,

Not currently, ProLIF only outputs residue-based IFPs so you can only compare bitvectors from the same protein family.

Most likely you want to look at ODDT's [PLEC ](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6477977/) fingerprint or [LUNA](https://github.com/keiserlab/LUNA) which produce bitvectors that encode atom-environments rather than residue information (and can then be folded to a fixed-length)

**Reply by jp43** [reply]
Thanks very much for your answer!
